src/js/editor/editor.js

```diff
@@ -677,6 +695,45 @@ class Editor {
    * @public
    */
   run(callback) {
+    return this._run(callback);
+  }
+  /**
+   * This does the actual work of the public {run} above, actionType is used to group undos.
+   * Currently there are two action types:
+   * 1) `EDITOR_RUN_LAST_ACTION_STATE.insert_text` which is used to group any text insertion, and
+   * 2) `EDITOR_RUN_LAST_ACTION_STATE.delete` which contains any delete event.
+   *
+   * Any other event that happens such as insertion of cards, atoms, or paste events creates their own undo
+   * entries and are not grouped.
+   *
+   * @param {Function} callback Called with an instance of
+   *        {@link PostEditor} as its argument.
+   * @param {Number|undefined} actionType provides the type of the current
+   *                                      action, but only if it's a "runnable"
+   *                                      type.
+   * @return {Mixed} The return value of `callback`.
+   * @private
+   */
+  _run(callback, actionType) {
```

Comment:
The reason that the state is being passed as an argument into the `run` method and not being set on the object is that `run` somehow needs to know that the command that it's executing is the command that set the state, because any event other than `insertText`, `deleteAtPosition`, and `deleteRange` should default to breaking the undo grouping.  I can't think of an elegant solution off the top of my head unfortunately.